rotkehlchen/api/v1/encoding.py

```diff
@@ -1109,6 +1113,52 @@ def _transform_eth_address(
     return address
 
 
+def _transform_ksm_address(
+        ethereum: EthereumManager,
+        given_address: str,
+) -> KusamaAddress:
+    """Returns a KSM address (if exists) given an ENS domain. At this point any
+    given address has been already validated either as an ENS name or as a
+    valid Kusama address (ss58 format).
+
+    NB: ENS domains for Substrate chains (e.g. KSM, DOT) store the Substrate
+    public key. It requires to encode it with a specific ss58 format for
+    obtainint the specific chain address.
+
+    Kusama/Polkadot ENS domain accounts:
+    https://guide.kusama.network/docs/en/mirror-ens
+
+    ENS domain substrate public key encoding:
+    https://github.com/ensdomains/address-encoder/blob/master/src/index.ts
+    """
+    if not given_address.endswith('.eth'):
+        return KusamaAddress(given_address)
+
+    resolved_address = cast(HexBytes, ethereum.ens_lookup(
+        given_address,
+        blockchain=SupportedBlockchain.KUSAMA,
+    ))
+    if resolved_address is None:
+        raise ValidationError(
+            f'Given ENS address {given_address} could not be resolved for Kusama',
+            field_name='address',
+        ) from None
+
+    public_key = SubstratePublicKey(resolved_address.hex())
+    try:
+        address = get_kusama_address_from_public_key(public_key)
+    except (TypeError, ValueError) as e:
+        raise ValidationError(
+            f"Given ENS address {given_address} does not contain a valid "
+            f"Substrate public key: {public_key}. Kusama address can't be obtained.",
+            field_name='address',
+        ) from e
+
+    log.info(f'Resolved ENS {given_address} to {address}')
```

Comment:
perhaps specify Resolved Kusama ENS and also this should be a debug log. No need to info this.